ui-v2/src/hooks/variables.ts

```diff
@@ -11,82 +11,91 @@ import {
+type Variables = UseVariablesOptions["variables"];
 
-type VariableKeys = {
-	all: readonly ["variables"];
-	filtered: (
-		options: UseVariablesOptions,
-	) => readonly ["variables", "filtered", string];
-	filteredCount: (options?: UseVariablesOptions) => readonly string[];
+// ----- 🌐 APIs 📨
+// ----------------------------
+const fetchVariablesRequest = async (body: UseVariablesOptions) => {
+	const response = await getQueryService().POST("/variables/filter", {
+		body,
+	});
+	return response.data;
 };
 
-/**
- * Query key factory for variables-related queries
- *
- * @property {readonly string[]} all - Base key for all variable queries
- * @property {function} filtered - Generates key for filtered variable queries, incorporating filter options
- * @property {function} filteredCount - Generates key for filtered count queries, including name and tag filters
- */
-const variableKeys: VariableKeys = {
-	all: ["variables"],
-	filtered: (options) => [
-		...variableKeys.all,
-		"filtered",
-		JSON.stringify(options),
-	],
-	filteredCount: (options) => {
-		const key = [...variableKeys.all, "filtered-count"];
-		if (options?.variables?.name?.like_) key.push(options.variables.name.like_);
-		if (options?.variables?.tags?.all_?.length)
-			key.push(JSON.stringify(options.variables.tags));
-		return key;
+const fetchVariableCountRequest = async (
+	body: UseVariablesOptions | Record<never, never>,
+) => {
+	const response = await getQueryService().POST("/variables/count", {
+		body,
+	});
+	return response.data;
+};
+
+const createVariableRequest = (
+	variable: components["schemas"]["VariableCreate"],
+) => {
+	return getQueryService().POST("/variables/", {
+		body: variable,
+	});
+};
+
+const updateVariableRequest = (
+	variable: components["schemas"]["VariableUpdate"] & {
+		id: string;
 	},
+) => {
+	const { id, ...body } = variable;
+	return getQueryService().PATCH("/variables/{id}", {
+		params: { path: { id } },
+		body,
+	});
 };
 
-/**
- * Builds a query configuration for fetching filtered variables
- * @param options - Filter options for the variables query including pagination, sorting, and filtering criteria
- * @returns A query configuration object compatible with TanStack Query including:
- *  - queryKey: A unique key for caching the query results
- *  - queryFn: The async function to fetch the filtered variables
- *  - staleTime: How long the data should be considered fresh (1 second)
- *  - placeholderData: Uses previous data while loading new data
- */
-const buildVariablesQuery = (options: UseVariablesOptions) =>
-	queryOptions({
-		queryKey: variableKeys.filtered(options),
-		queryFn: async () => {
-			const response = await getQueryService().POST("/variables/filter", {
-				body: options,
-			});
-			return response.data;
-		},
-		staleTime: 1000,
-		placeholderData: keepPreviousData,
+const deleteVariableRequest = (id: string) => {
+	return getQueryService().DELETE("/variables/{id}", {
+		params: { path: { id } },
```

Comment:
Since these are simple functions that are each only used in one place, I'd prefer to inline them in the query options where they are used. 

I suspect this change is related to this line in the PR description:

> Abstract queryFn and mutationFn into its own function wrapper. Purpose is to separate concerns from APIs and queries 🧅. Additionally most https://github.com/TanStack docs [follow a similar pattern](https://tanstack.com/router/latest/docs/framework/react/examples/basic-react-query-file-based)

Our query service already owns the nuts and bolts of API calls. These functions are thin adapters between the query service and the query/mutation hooks, which is also why I think they should remain anonymous functions.

Reply:
Reverted back in: https://github.com/PrefectHQ/prefect/pull/16155/commits/d8d505e546b66710bc1c8ef7dfc006d0602a338c